frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -632,6 +693,23 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
 
             router.actions.push(urls.insightView(insight.short_id))
         },
+        updateInsight: async () => {
+            if (!values.editingInsight) {
+                return
+            }
+
+            const insightName = values.activeModelUri?.name
+
+            const insightRequest: Partial<QueryBasedInsightModel> = {
+                name: insightName ?? values.editingInsight.name,
+                query: values.sourceQuery,
+            }
+
+            const savedInsight = await insightsApi.update(values.editingInsight.id, insightRequest)
+
+            lemonToast.info(`You're now viewing ${savedInsight.name || savedInsight.derived_name || name}`)
+            router.actions.push(urls.insightView(savedInsight.short_id))
+        },
```

Comment:
This doesn't give me an updated state when I'm back on the view insight tab. (e.g. if i have a viz showing value 2 and update it to 3 when it directs me to the insight view it's not 3 and still 2)